### PR TITLE
refactor(core): more descriptive unknown element exception

### PR DESCRIPTION
**Pull request by andreimc**
providing the element name would help you pin-point your missing provider quicker

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: packages/core/injector/container-scanner.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class ContainerScanner {
       : metatypeOrToken;
     const instanceWrapper = dependencies.get(name as string);
     if (!instanceWrapper) {
-      throw new UnknownElementException();
+      throw new UnknownElementException(name as string);
```
Comment (BrunnerLivio) [inline packages/core/injector/container-scanner.ts]:
```suggestion
      throw new UnknownElementException(name.toString());
```

`name` may be a symbol, therefore you need to call `toString()` so the token will get display as `Symbol(MY_TOKEN)`. #1551

File: packages/core/errors/exceptions/unknown-element.exception.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { RuntimeException } from './runtime.exception';
 
 export class UnknownElementException extends RuntimeException {
-  constructor() {
+  constructor(name?: string) {
     super(
-      'Nest cannot find given element (it does not exist in current context)',
+      `Nest cannot find ${name ? name : 'given'} element (it does not exist in current context)`,
```
Comment (BrunnerLivio) [inline packages/core/errors/exceptions/unknown-element.exception.ts]:
```suggestion
      `Nest cannot find ${name || 'given'} element (it does not exist in current context)`,
```

**Comment by BrunnerLivio** [comment]
@andreimc Great PR, thanks a lot!

**Comment by andreimc** [comment]
> @andreimc Great PR, thanks a lot!

thanks, updated the PR with your changes :D!

File: packages/core/injector/container-scanner.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export class ContainerScanner {
       : metatypeOrToken;
     const instanceWrapper = dependencies.get(name as string);
     if (!instanceWrapper) {
-      throw new UnknownElementException();
+      throw new UnknownElementException(name.toString());
```
Comment (kamilmysliwiec) [inline packages/core/injector/container-scanner.ts]: Should we add an extra check against `null` / `undefined` values here? `name && name.toString()`

Comment (BrunnerLivio) [reply packages/core/injector/container-scanner.ts]:
```suggestion
      throw new UnknownElementException(name && name.toString());
```

so @andreimc can accept it easily ;)

**Comment by andreimc** [comment]
Updated :)

**Pull request merged**